### PR TITLE
New plugin: send errors to Sentry

### PR DESCRIPTION
**Pull request by filippog**
##### SUMMARY
This plugin will send errors from Ansible to Sentry, including timeouts and Python exceptions.

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
sentry callback plugin

### CODE REVIEW

File: lib/ansible/plugins/callback/sentry.py [hunk 1]
```diff
@@ -0,0 +1,108 @@
...
+
+__metaclass__ = type
+
+DOCUMENTATION = """
+    callback: sentry
+    type: notification
+    short_description: Sends errors to Sentry
+    description:
+      - Send Ansible errors and exceptions to Sentry
+    version_added: "2.7"
```
Comment (gundalow) [inline lib/ansible/plugins/callback/sentry.py]: 2.8

File: lib/ansible/plugins/callback/sentry.py [hunk 2]
```diff
@@ -0,0 +1,108 @@
...
+    CALLBACK_TYPE = "aggregate"
+    CALLBACK_NAME = "sentry"
+    CALLBACK_NEEDS_WHITELIST = True
+
+    def __init__(self):
+        super(CallbackModule, self).__init__()
+
+        if not HAS_SENTRY:
+            self.disabled = True
+            self._display.warning("Package 'sentry-sdk' not found.")
```
Comment (bcoca) [inline lib/ansible/plugins/callback/sentry.py]: a more meaningful message would be helpful: 'The python library 'sentry-sdk' was not found and is required for the sentry callback plugin to run, so the plugin is disabled for this run.'

File: lib/ansible/plugins/callback/sentry.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+class CallbackModule(CallbackBase):
+    """
+    Requires:
+        sentry-sdk
+
+    Environment:
+        SENTRY_DSN (required): Sentry DSN
+    """
+
+    CALLBACK_VERSION = 0.1
```
Comment (bcoca) [inline lib/ansible/plugins/callback/sentry.py]: this is not the version of the callback itself, but the API, 2.0 would be the correct number for this usage

**Review by bcoca — CHANGES_REQUESTED** [review]

File: lib/ansible/plugins/callback/sentry.py (hunk 1 above)
Comment (filippog) [reply lib/ansible/plugins/callback/sentry.py]: Fixed!

File: lib/ansible/plugins/callback/sentry.py (hunk 2 above)
Comment (filippog) [reply lib/ansible/plugins/callback/sentry.py]: Makes sense, I've changed the message.

File: lib/ansible/plugins/callback/sentry.py (hunk 3 above)
Comment (filippog) [reply lib/ansible/plugins/callback/sentry.py]: Whoops! Fixed.

**Comment by filippog** [comment]
Thanks for the comments, addressed them now, PTAL!

**Comment by Akasurde** [comment]
Hi @filippog, Thank you very much for your interest in Ansible. This plugin/module is no longer maintained in this repository and has been migrated to https://github.com/ansible-collections/community.general<br><br>If you have further questions please stop by IRC or the mailing list:<br><br>* IRC: #ansible on irc.freenode.net<br>* mailing list: https://groups.google.com/forum/#!forum/ansible-project<br><br>needs_info<br>

**Pull request closed without merging**